plugins/out_azure_kusto/azure_kusto_ingest.c

```diff
@@ -298,145 +349,149 @@ static int azure_kusto_enqueue_ingestion(struct flb_azure_kusto *ctx, flb_sds_t
+
+                            if (c->resp.payload_size > 0) {
+                                flb_plg_debug(ctx->ins,
+                                              "Request failed and returned: \n%s",
+                                              c->resp.payload);
+                            }
+                            else {
+                                flb_plg_debug(ctx->ins, "Request failed");
+                            }
+                        }
+                    }
+                    else {
+                        flb_plg_error(ctx->ins, "cannot send HTTP request");
+                    }
+
+                    flb_http_client_destroy(c);
+                }
+                else {
+                    flb_plg_error(ctx->ins,
+                                  "cannot create HTTP client context for queue");
+                }
+
+                flb_sds_destroy(payload);
+            }
+            else {
+                flb_plg_error(ctx->ins, "error creating payload buffer");
+            }
+
+            flb_sds_destroy(uri);
+        }
+        else {
+            flb_plg_error(ctx->ins, "error creating queue uri buffer");
+        }
 
-    payload = create_ingestion_message(ctx, blob_uri, payload_size);
-    if (!payload) {
-        flb_plg_error(ctx->ins, "error creating payload buffer");
         flb_upstream_conn_release(u_conn);
-        flb_sds_destroy(uri);
-        return -1;
     }
-
-    c = flb_http_client(u_conn, FLB_HTTP_POST, uri, payload, flb_sds_len(payload), NULL,
-                        0, NULL, 0);
-    if (!c) {
-        flb_plg_error(ctx->ins, "cannot create HTTP client context for blob container");
-        flb_upstream_conn_release(u_conn);
-        flb_sds_destroy(uri);
-        flb_sds_destroy(payload);
-        return -1;
+    else {
+        flb_plg_error(ctx->ins, "error getting queue upstream connection");
     }
 
-    flb_http_add_header(c, "User-Agent", 10, "Fluent-Bit", 10);
-    flb_http_add_header(c, "Content-Type", 12, "application/atom+xml", 20);
-    flb_http_add_header(c, "x-ms-date", 9, tmp, len);
-    flb_http_add_header(c, "x-ms-version", 12, "2019-12-12", 10);
+    return ret;
+}
 
-    ret = flb_http_do(c, &resp_size);
-    flb_plg_debug(ctx->ins, "kusto queue request http_do=%i, HTTP Status: %i", ret,
-                  c->resp.status);
-    if (ret != 0) {
-        flb_plg_error(ctx->ins, "cannot send HTTP request");
-        flb_upstream_conn_release(u_conn);
-        flb_sds_destroy(uri);
-        flb_sds_destroy(payload);
-        flb_http_client_destroy(c);
-        return -1;
-    }
+static flb_sds_t azure_kusto_create_blob_id(struct flb_azure_kusto *ctx, flb_sds_t tag,
+                                            size_t tag_len)
+{
+    flb_sds_t blob_id = NULL;
+    struct flb_time tm;
+    uint64_t ms;
+    char *b64tag;
+    size_t b64_len;
+
+    flb_time_get(&tm);
+    ms = ((tm.tm.tv_sec * 1000) + (tm.tm.tv_nsec / 1000000));
```

Comment:
We should create `flb_time_to_milliseconds` or so plugins don't have to repeat that code but that's not a complaint about this, it's a to-do for us =)